eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/JavadocCodeSnippetCheck.java

```diff
@@ -0,0 +1,256 @@
+                }
+                break;
+            case TokenTypes.METHOD_DEF:
+                if (!methodDefStack.isEmpty()) {
+                    methodDefStack.pop();
+                }
+                break;
+            default:
+                // Checkstyle complains if there's no default block in switch
+                break;
+        }
+    }
+
+    @Override
+    public void visitToken(DetailAST token) {
+        switch (token.getType()) {
+            case TokenTypes.PACKAGE_DEF:
+                packageName = FullIdent.createFullIdent(token.findFirstToken(TokenTypes.DOT)).getText();
+                break;
+            case TokenTypes.CLASS_DEF:
+                classNameStack.push(token.findFirstToken(TokenTypes.IDENT).getText());
+                break;
+            case TokenTypes.METHOD_DEF:
+                methodDefStack.push(token);
+                break;
+            case TokenTypes.BLOCK_COMMENT_BEGIN:
+                checkNamingPattern(token);
+                break;
+            default:
+                // Checkstyle complains if there's no default block in switch
+                break;
+        }
+    }
+
+    /**
+     * Check if the given block comment is on method. If not, skip the check.
+     * Otherwise, check if the codesnippet has matching the naming pattern
+     *
+     * @param blockCommentToken BLOCK_COMMENT_BEGIN token
+     */
+    private void checkNamingPattern(DetailAST blockCommentToken) {
+        if (!BlockCommentPosition.isOnMethod(blockCommentToken)) {
+            return;
+        }
+
+        // Turn the DetailAST into a Javadoc DetailNode.
+        DetailNode javadocNode = null;
+        try {
+            javadocNode = DetailNodeTreeStringPrinter.parseJavadocAsDetailNode(blockCommentToken);
+        } catch (IllegalArgumentException ex) {
+            // Exceptions are thrown if the JavaDoc has invalid formatting.
+        }
+
+        if (javadocNode == null) {
+            return;
+        }
+
+        // Iterate through all the top level nodes in the Javadoc, looking for the @throws statements.
+        for (DetailNode node : javadocNode.getChildren()) {
+            if (node.getType() != JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                continue;
+            }
+            // Skip if not codesnippet
+            DetailNode customNameNode = JavadocUtil.findFirstToken(node, JavadocTokenTypes.CUSTOM_NAME);
+            if (customNameNode == null || !CODE_SNIPPET_ANNOTATION.equals(customNameNode.getText())) {
+                return;
+            }
+            // Missing Description
+            DetailNode descriptionNode = JavadocUtil.findFirstToken(node, JavadocTokenTypes.DESCRIPTION);
+            if (descriptionNode == null) {
+                log(node.getLineNumber(), MISSING_CODESNIPPET_TAG_MESSAGE);
+                return;
+            }
+
+            String customDescription = JavadocUtil.findFirstToken(descriptionNode, JavadocTokenTypes.TEXT).getText();
```

Comment:
Will this result in a NullReferenceException if you do not have text for your description node.. findFirstToken returns null? (ie. `{@codesnippet}`).

Reply:
There will always have TEXT token if there is DESCRIPTION token exists. Will add this comment to the above of this code section